paddle/fluid/framework/operator.cc

```diff
@@ -1175,6 +1175,34 @@ proto::VarType::Type OperatorWithKernel::IndicateDataType(
   return data_type;
 }
 
+proto::VarType::Type OperatorWithKernel::IndicateVarDataType(
+    const ExecutionContext& ctx, const std::string& name) const {
+  const std::vector<const Variable*> vars = ctx.MultiInputVar(name);
+  PADDLE_ENFORCE_EQ(
+      vars.empty(), false,
+      "The %s Op doesn't have Input %s or it is an empty Variable.",
+      this->type_, name);
+  const Variable* var = vars.front();
```

Comment:
Why get `front()` only?

Reply:
currently, the `MultiInputVar` used in `GetExpectedKernelType`only has following two forms:
`ctx.MultiInputVar("X")[0]`
`ctx.MultiInputVar("Ids").front()`
So I only get `front()`, but it is hard, already fixed